tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +145,147 @@ def get_rdm(self):
+                                                    frozen_orbitals=molecule.frozen_orbitals)
+        self.basis = molecule.basis
+
+    def simulate(self):
+        """Perform the simulation (energy calculation) for the molecule.
+
+        Returns:
+            float: Total FCI energy.
+        """
+        n_frozen_vir = len(self.molecule.frozen_virtual)
+        n_frozen_occ = len(self.molecule.frozen_occupied)
+        ref = 'rhf' if self.molecule.spin == 0 else 'rohf'
+        self.backend.set_options({'basis': self.basis, 'mcscf_maxiter': 300, 'mcscf_diis_start': 20,
+                                  'opdm': True, 'tpdm': True, 'frozen_docc': [n_frozen_occ], 'frozen_uocc': [n_frozen_vir],
+                                  'qc_module': 'detci', 'fci': True, 'reference': ref})
+
+        # Copy reference wavefunction and swap orbitals to obtain correct active space if necessary
+        wfn = self.backend.core.Wavefunction(self.molecule.solver.mol_nosym, self.molecule.solver.wfn.basisset())
+        wfn.deep_copy(self.molecule.solver.wfn)
+        if n_frozen_occ or n_frozen_vir:
+            mo_order = self.molecule.frozen_occupied + self.molecule.active_occupied + self.molecule.active_virtual + self.molecule.frozen_virtual
+            # Obtain swap operations that will take the unordered list back to ordered with the correct active space in the middle.
+            swap_ops = Permutation(mo_order).transpositions()
+            for swap_op in swap_ops[::-1]:
+                wfn.Ca().rotate_columns(0, swap_op[0], swap_op[1], np.deg2rad(90))
+
+        energy, self.ciwfn = self.backend.energy('fci', molecule=self.molecule.solver.mol,
+                                                 basis=self.basis, return_wfn=True,
+                                                 ref_wfn=wfn)
+        return energy
+
+    def get_rdm(self):
+        """Compute the Full CI 1- and 2-particle reduced density matrices.
+
+        Returns:
+            numpy.array: One-particle RDM.
+            numpy.array: Two-particle RDM.
+
+        Raises:
+            RuntimeError: If method "simulate" hasn't been run.
+        """
+
+        # Check if Full CI has been performed
+        if self.ciwfn is None:
+            raise RuntimeError("FCISolver: Cannot retrieve RDM. Please run the 'simulate' method first")
+        if self.degenerate_mo_energies:
+            warnings.warn("Degenerate orbitals are present in the molecule. The fci calculation is performed "
+                          "without symmetry so the rdms may not correspond to the integrals in molecule. A c1 "
+                          "version of the molecule with the correct integrals is present as FCISolverPsi4.molecule.")
+        one_rdm = np.asarray(self.ciwfn.get_opdm(0, 0, "SUM", False))
+        two_rdm = np.asarray(self.ciwfn.get_tpdm("SUM", True))
+
+        return one_rdm, two_rdm
+
+
+fci_solver_dict = {"pyscf": FCISolverPySCF, 'psi4': FCISolverPsi4}
```

Comment:
this seems like a very random place for this variable.

Reply:
It throws an error if I put it at the top of the file. Apparently python cares within a file.

Reply:
Then I stand corrected :)

Minor but you are not being consistent with `'` and `"` in your dictionary keys.